Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -15,8 +17,32 @@ import NIOTransportServices
 /// Convenience shorthand for `EventLoopFuture`.
 public typealias Future = EventLoopFuture
 
+/// Helper struct for ensuring atomic access to a structure
+struct AtomicProperty<T> {
```

Comment:
I don't like this here. Maybe we can work around this with the approach in: https://github.com/adam-fowler/aws-signer/pull/2

Reply:
I understand your concerns about this, but I think the changes you have suggested are too big. See my version in the comments of your PR.